specs/~dynamic-names.yml

```
@@ -0,0 +1,349 @@
+overview: |
+  Rationale: this special notation was introduced primarily to allow the dynamic
+  loading of partials. The main advantage that this notation offers is to allow
+  dynamic loading of partials, which is particularly useful in cases where
+  polymorphic data needs to be rendered in different ways, cases which would
+  otherwise be possible to render only with solutions that are convoluted,
+  inefficient, or both.
+  Example.
+  Let's consider the following data:
+    items: [
+      { content: 'Hello, World!' },
+      { url: 'http://example.com/foo.jpg' },
+      { content: 'Some text' },
+      { content: 'Some other text' },
+      { url: 'http://example.com/bar.jpg' },
+      { url: 'http://example.com/baz.jpg' },
+      { content: 'Last text here' }
+    ]
+  The goal is to render the different types of items in different ways: the
+  items having a key named `content` should be rendered with the template
+  `text.mustache` and the items having a key named `url` should be rendered
+  with the template `image.mustache`:
+  text.mustache:
+    {{!image.mustache}}
+    <img src="{{url}}"/>
+  image.mustache:
+    {{!text.mustache}}
+    {{content}}
+  There are already several ways to achieve this goal, here below are
+  illustrated and discussed the most significant solutions to this problem.
+  ## Using Pre-Processing
+  The idea is to use a secondary templating mechanism to dynamically generate
+  the template that will be rendered.
+  The template that our secondary templating mechanism generates should look
+  like this:
+    {{!template.mustache}}
+    {{items.1.content}}
+    <img src="{{items.2.url}}"/>
+    {{items.3.content}}
+    {{items.4.content}}
+    <img src="{{items.5.url}}"/>
+    <img src="{{items.6.url}}"/>
+    {{items.7.content}}
+  This solutions offers the advantages of having more control over the template
+  and minimizing the template blocks to the essential ones.
+  The drawbacks are the rendering speed and the complexity that the secondary
+  templating mechanism requires.
+  ## Using Lambdas
+  The idea is to inject into the data functions that will be later called from
```

```suggestion
  The idea is to inject functions into the data that will be later called from
```

This order seems more natural to me. I'm only a B2 though, not a native speaker.

---

Agreed. I'm not a native speaker either.